datafusion/physical-expr/src/equivalence.rs

```diff
@@ -885,6 +885,240 @@ fn req_satisfied(given: LexOrderingRef, req: &[PhysicalSortRequirement]) -> bool
+                for left_ordering in left_oeq_class.iter() {
+                    for right_ordering in updated_right_oeq.iter() {
+                        let mut ordering = left_ordering.to_vec();
+                        ordering.extend(right_ordering.to_vec());
+                        let ordering_normalized =
+                            join_eq_properties.normalize_sort_exprs(&ordering);
+                        orderings.push(ordering_normalized);
+                    }
+                }
+                if !orderings.is_empty() {
+                    let head = orderings.swap_remove(0);
+                    let new_oeq_class = OrderingEquivalentClass::new(head, orderings);
+                    new_properties.extend(Some(new_oeq_class));
+                }
+            } else {
+                new_properties.extend(left_oeq_properties.oeq_class().cloned());
+            }
+        }
+        (false, true) => {
+            let updated_right_oeq = right_oeq_properties
+                .oeq_class()
+                .map(|right_oeq_class| {
+                    get_updated_right_ordering_equivalent_class(
+                        join_type,
+                        right_oeq_class,
+                        left_columns_len,
+                        &join_eq_properties,
+                    )
+                })
+                .transpose()?;
+            // In this special case, left side ordering can be prefixed with right side ordering.
+            if let (
+                Some(JoinSide::Right),
+                JoinType::Inner,
+                Some(left_oeq_class),
+                Some(right_oeg_class),
+            ) = (
+                probe_side,
+                join_type,
+                left_oeq_properties.oeq_class(),
+                &updated_right_oeq,
+            ) {
+                // Left side ordering equivalence properties should be prepended with
+                // those of the right side while constructing output ordering equivalence
+                // properties since stream side is the right side.
+                //
+                // If the right table ordering equivalences contain `b ASC`, and the output
+                // ordering of the left table is `a ASC`, then the ordering equivalence `b ASC`
+                // for the right table should be converted to `a ASC, b ASC` before it is added
+                // to the ordering equivalences of the join.
+                let mut orderings = vec![];
+                for right_ordering in right_oeg_class.iter() {
+                    for left_ordering in left_oeq_class.iter() {
+                        let mut ordering = right_ordering.to_vec();
+                        ordering.extend(left_ordering.to_vec());
+                        let ordering_normalized =
+                            join_eq_properties.normalize_sort_exprs(&ordering);
+                        orderings.push(ordering_normalized);
+                    }
+                }
```

Comment:
Similar to above comment. However, in this case left orderings are prefixed with right orderings